internal/util/schemaupdate/schemaupdate.go

```diff
@@ -119,7 +135,7 @@ func validateProperties(oldSchemaMap, newSchemaMap map[string]any) error {
 
 	// The new schema should be a superset of the old schema
 	// if it's not, we may break profiles using this rule type
-	if !cmp.Equal(newSchemaMap, castedDst, opts...) {
+	if !cmp.Equal(newPropertiesMap, castedDst, opts...) {
```

Comment:
hey, do I read this correct that `castedDst` comes from `dst` which comes from `newPropertiesMap`? If so, perhaps the comment on top might not be valid